Lib/_collections_abc.py

```diff
@@ -1013,20 +1013,10 @@ def index(self, value, start=0, stop=None):
            Supporting start and stop arguments is optional, but
            recommended.
         '''
-        if start is not None and start < 0:
-            start = max(len(self) + start, 0)
-        if stop is not None and stop < 0:
-            stop += len(self)
-
-        i = start
-        while stop is None or i < stop:
-            try:
-                v = self[i]
-                if v is value or v == value:
-                    return i
-            except IndexError:
-                break
-            i += 1
+        for i in range(len(self))[start:stop]:
```

Comment:
This code loses the handling for a negative start value.

The reason this code looks messy is that it follows the implementation of list_index.

Reply:
> This code loses the handling for a negative start value.

Are you sure that `range(len(self))[start:stop]` (which is equivalent to `range(*slice(start, stop).indices(len(self)))` by the way) does not handle negative start values? Negative indices in slicing [start at the last item](https://stackoverflow.com/questions/509211/understanding-slicing/71640042#71640042) (so `len(self) + start`), and since we slice `range(len(self))` we also clip at `0`. In other words the handling of negative values is equivalent to `start = max(len(self) + start, 0)`. Or am I missing something?

> The reason this code looks messy is that it follows the implementation of list_index.

Okay. So I am fine with keeping it like that way. The only thing that still bothers me is the `if` statement in the `try` clause since it cannot raise `IndexError`. Can we move it after the `try` statement? Like this:

```python
        while stop is None or i < stop:
             try:
                 v = self[i]
             except IndexError:
                 break
             if v is value or v == value:
                 return i
             i += 1
```